lib/codecs/src/decoding/format/avro.rs

```diff
@@ -0,0 +1,302 @@
+use super::Deserializer;
+use crate::encoding::AvroSerializerOptions;
+use bytes::Buf;
+use bytes::Bytes;
+use lookup::event_path;
+use serde::{Deserialize, Serialize};
+use smallvec::{smallvec, SmallVec};
+use vector_config::configurable_component;
+use vector_core::{
+    config::{log_schema, DataType, LogNamespace},
+    event::{Event, LogEvent},
+    schema,
+};
+
+type VrlValue = vrl::value::Value;
+type AvroValue = apache_avro::types::Value;
+
+const CONFLUENT_MAGIC_BYTE: u8 = 0;
+const CONFLUENT_SCHEMA_PREFIX_LEN: usize = 5;
+
+/// Config used to build a `AvroDeserializer`.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct AvroDeserializerConfig {
+    /// Options for the Avro deserializer.
+    pub avro_options: AvroDeserializerOptions,
+}
+
+impl AvroDeserializerConfig {
+    /// Creates a new `AvroDeserializerConfig`.
+    pub const fn new(schema: String, strip_schema_id_prefix: bool) -> Self {
+        Self {
+            avro_options: AvroDeserializerOptions {
+                schema,
+                strip_schema_id_prefix,
+            },
+        }
+    }
+
+    /// Build the `AvroDeserializer` from this configuration.
+    pub fn build(&self) -> AvroDeserializer {
+        let schema = apache_avro::Schema::parse_str(&self.avro_options.schema)
+            .map_err(|error| format!("Failed building Avro serializer: {}", error))
+            .unwrap();
+        AvroDeserializer {
+            schema,
+            strip_schema_id_prefix: self.avro_options.strip_schema_id_prefix,
+        }
+    }
+
+    /// The data type of events that are accepted by `AvroDeserializer`.
+    pub fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    /// The schema required by the serializer.
+    pub fn schema_definition(&self, log_namespace: LogNamespace) -> schema::Definition {
+        // TODO: convert avro schema to vector schema.
```

Comment:
💬 Do you plan to change this?

Reply:
No plans in the near future, probably after finishing the other task at my hand. Do I need to remove the `TODO` comment?

Reply:
Feel free to remove or edit the comment. I would like to understand better the intention here. Can you explain a bit more?

Reply:
Here it is intended to convert AVRO schema definitions to vectors for checksumming by other modules similar to what syslog does. However others decoders like protobuf seem to be simple, so I removed the todo tag here by 
[9aeab12.](https://github.com/vectordotdev/vector/pull/18636/commits/9ff71550c4b9c087a56ccc01e3280aeadca889bc)